src/sentry/incidents/logic.py

```diff
@@ -1557,22 +1627,31 @@ def get_alert_rule_trigger_action_slack_channel_id(
     return channel_data.channel_id
 
 
-def get_alert_rule_trigger_action_discord_channel_id(
-    name,
-    organization,
-    integration_id,
-    use_async_lookup=False,
-    input_channel_id=None,
-    integrations=None,
-):
+def _get_alert_rule_trigger_action_discord_channel_id(
+    name: str | None, integration_id: int
+) -> str | None:
     from sentry.integrations.discord.utils.channel import validate_channel_id
 
     integration = integration_service.get_integration(integration_id=integration_id)
     if integration is None:
         raise InvalidTriggerActionError("Discord integration not found.")
     try:
+        # TODO(RyanSkonnord): Force this function's `name` param to be non-null.
+        #
+        # AlertRuleTriggerActionSerializerTest.test_discord_channel_id_none covers
+        # the case of `name` being None explicitly. However, from eyeballing
+        # validate_channel_id, it looks like this can only lead to an error. The unit
+        # test case originates in https://github.com/getsentry/sentry/pull/58005,
+        # where it looks like it was targeting a spurious failure mode that occurs
+        # while building the description string.
+        #
+        # I'd like to move the null check upstream, clean up the associated nullable
+        # function params, remove the piecemeal checks from
+        # _get_target_identifier_display_from_target_value, and modify
+        # test_discord_channel_id_none to assert for an explicit null check (or just
+        # delete it).
```

Comment:
Assuming no one objects to modifying the test case, I'd be ready to make these changes right away. I just didn't want to do it in the same PR.